spack/packages/dla-future/package.py

```diff
@@ -16,23 +15,41 @@ class DlaFuture(CMakePackage):
 
     version('develop', branch='master')
 
-    variant('gpu', default=False,
+    variant('cuda', default=False,
             description='Use the GPU/cuBLAS back end.')
-
-    # Until mpich is default comment this out
+    variant('test', default=False,
+            description='Run unit tests.')
```

Comment:
I'm not really sure it is the right way of doing it.

See:
- [Build-time tests doc](https://spack.readthedocs.io/en/latest/packaging_guide.html#build-time-tests)
- [CMakePackage doc](https://spack.readthedocs.io/en/latest/build_systems/cmakepackage.html?highlight=cmakepackage#testing)

Reply:
Yes, this variant is not needed, users can control testing with `spack install --test=root`. We can remove the variant and the `+test` check in `cmake_args()` and replace it with this:

```
args = [-DDLAF_WITH_TEST'={0}'.format('ON' if self.run_tests else 'OFF'), -DDLAF_INSTALL_TESTS'={0}'.format('ON' if self.run_tests else 'OFF')]
```

Reply:
true, so that we can run the install with `--run_tests`. @sanciop and I remove the variant `+test` and modified the argument accordingly. is it ok?

Reply:
Yes, this is OK!